### PR TITLE
PR to merge Rspec to master

### PR DESCRIPTION
**Pull request by Haroonabdulrazaq**
- Add test for Board method
- Add test for Switch_Player method
- Add test for Update_board method
- Add test for the win? method

### CODE REVIEW

File: spec/game_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    it 'returns the board string' do
+      my_result = game.board
+      expect(my_result).to eql(str)
+    end
+  end
+
+  describe '#switch_player' do
+    it 'returns player1 or Player 2' do
+      expect(game.switch_player).to eql(1)
+      expect(game.switch_player).to eql(0)
```
Comment (maelfosso) [inline spec/game_spec.rb]: if you switch player, we expect you to check not the return value but the variable saving  the current player. In your case `@player`

File: spec/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    end
+  end
+
+  describe '#update_board' do
+    it 'returns a mark to update the board with O' do
+      expect(game.update_board(5)).to eql(game.player.mark)
+    end
```
Comment (maelfosso) [inline spec/game_spec.rb]: If you update the board, we expect you to check the directly the board. In that situation, you have to check `@position[5]`

File: spec/game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      it "returns true if player occupies #{i.map { |j| j + 1 }}" do
+        game.position[i[0]] = game.player.mark
+        game.position[i[1]] = game.player.mark
+        game.position[i[2]] = game.player.mark
+        expect(game.win?).to eql(true)
+      end
+    end
+    it 'returns false if player doesn\'t occupy any winning move' do
+      expect(!game.win?).to eql(false)
+    end
```
Comment (maelfosso) [inline spec/game_spec.rb]: To test that method, you select cases, simulate a game that is an update board and after some iterations check whether the expected result of the `win?` method.

Be sure to alternate true/false result

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for the effort put in this milestone.
Please, put a bit more effort to fix some issues. Read the comments left below.

Once, changes done, kindly submit a new code review request
Thanks

Mael FOSSO

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

###### PS: Using `require_relative` for files in your directory instead of `require` is a ruby convention.

> "Require should be used for external files, like gems, while require_relative should be used for referring to files within your directory."

**Pull request merged**